src/app/component/navbar.tsx

```diff
@@ -1,90 +1,149 @@
+import type { Route } from 'next';
 
- 
 const Navbar: FC = () => {
+	const linkGeneral = [
+		{
+			label: 'Dashboard',
+			route: '/',
+			ihov: '\navbardashboard-purple.svg',
+			i: '\navbardashboard-colorless.svg',
+		},
+		{
+			label: 'Message',
+			route: '/message',
+			ihov: '\navbarmessage-purple.svg',
+			i: '\navbarmessage-colorless.svg',
+		},
+		{
+			label: 'settings',
+			route: '/settings',
+			ihov: '\navbarsettings-purple.svg',
+			i: '\navbarsettings-colorless.svg',
+		},
+	];
 
-    const linkGeneral =[
-        {label:'Dashboard', route:'/', ihov:"\navbar\dashboard-purple.svg", i:"\navbar\dashboard-colorless.svg",},
-        {label:'Message', route:'/message', ihov:"\navbar\message-purple.svg", i:"\navbar\message-colorless.svg",},
-        {label:'settings', route:'/settings', ihov:"\navbar\settings-purple.svg", i:"\navbar\settings-colorless.svg",},        
-    ]
+	const linkMarketplace = [
+		{
+			label: 'market',
+			route: '/market',
+			ihov: '\navbarmarket-purple.svg',
+			i: '\navbarmarket-colorless.svg',
+		},
+		{
+			label: 'mypositions',
+			route: '/mypositions',
+			ihov: '\navbarmypositions-purple.svg',
+			i: '\navbarmypositions-colorless.svg',
+		},
+	];
 
-    const linkMarketplace =[
-        {label:'market', route:'/market', ihov:"\navbar\market-purple.svg", i:"\navbar\market-colorless.svg",},
-        {label:'mypositions', route:'/mypositions', ihov:"\navbar\mypositions-purple.svg", i:"\navbar\mypositions-colorless.svg",},
-    ]
+	const linkMyprof = [
+		{
+			label: 'collection',
+			route: '/collection',
+			ihov: '\navbarcollection-purple.svg',
+			i: '\navbarcollection-colorless.svg',
+		},
+		{
+			label: 'wallet',
+			route: '/wallet',
+			ihov: '\navbarwallet-purple.svg',
+			i: '\navbarwallet-colorless.svg',
+		},
+		{
+			label: 'history',
+			route: '/history',
+			ihov: '\navbarhistory-purple.svg',
+			i: '\navbarhistory-colorless.svg',
+		},
+	];
 
+	const router = useRouter();
+	const pathName = usePathname();
+	const [active, setActive] = useState(pathName);
+	const [isHovering, setIsHovering] = useState(false);
+	const [hoverActive, setHoverActive] = useState('');
+	const handleMouseOver = (label: string) => {
+		setIsHovering(true);
+		setHoverActive(label);
+	};
+	const handleMouseOut = () => {
+		setIsHovering(false);
+		setHoverActive('');
+	};
 
-    const linkMyprof =[
-        {label:'collection', route:'/collection', ihov:"\navbar\collection-purple.svg", i:"\navbar\collection-colorless.svg",},
-        {label:'wallet', route:'/wallet', ihov:"\navbar\wallet-purple.svg", i:"\navbar\wallet-colorless.svg",},
-        {label:'history', route:'/history', ihov:"\navbar\history-purple.svg", i:"\navbar\history-colorless.svg",},         
-    ]
+	return (
+		<>
+			<nav>
+				<h3>General</h3>
+				<ul>
+					<li>
+						<Link href={'/' as Route}>
+							<Image src="\navbar\history-colorless.svg" alt="Setting icon" />{' '}
+							Dashboard
+						</Link>
+					</li>
```

Comment:
There is a lot of code duplication here
Add a component on the top to wrap the IU and pass the rest as a loop, make an object with the different options (image, title)

Reply:
Remove duplicate code